ambry-server/src/main/java/com/github/ambry/server/AmbryServerRequests.java

```diff
@@ -439,14 +450,39 @@ private AdminResponse handleHealthCheckRequest(DataInputStream requestStream, Ad
         if (hostBlobStore == null || !hostBlobStore.isStarted() ||
             ((hostBlobStore.getCurrentState() != ReplicaState.STANDBY) &&
                 (hostBlobStore.getCurrentState() != ReplicaState.LEADER))) {
-          hostHealthy = false;
+          serverHealthStatus = ServerHealthStatus.BAD;
           break;
         }
       }
+
+      /*
+       Checks if every disk that's assigned to a partition is also mounted and disk is healthy
+       */
+      HashSet<DiskManager> partitionsDiskManagers = new HashSet<>(storageManager.getPartitionToDiskManager().values());
+      for(Map.Entry<DiskId,DiskManager> entry : storageManager.getDiskToDiskManager().entrySet()){
+        DiskManager mountedDiskManager = entry.getValue();
+
+        //Checks Disk healthchecking is enabled and
+        // if a disk isn't healthy then store that disk to return in the response
+        if(diskManagerConfig.diskManagerDiskHealthCheckEnabled &&
+            (!partitionsDiskManagers.contains(mountedDiskManager) ||
+            mountedDiskManager.getDiskHealthStatus() != DiskHealthStatus.HEALTHY)){
+
+          serverHealthStatus = ServerHealthStatus.BAD;
+          JSONObject brokenDisk = new JSONObject();
+          brokenDisk.put("disk",entry.getKey().getMountPath());
+          brokenDisk.put("reason",mountedDiskManager.getDiskHealthStatus());
+          brokenDisks.put(brokenDisk);
+        }
+      }
```

Comment:
we can put `diskManagerConfig.diskManagerDiskHealthCheckEnabled` outside of this for loop.

Reply:
Thanks. Changed it